lib/solr_configurator.rb

```diff
@@ -0,0 +1,129 @@
+# frozen_string_literal: true
+
+require 'zip'
+
+class SolrConfigurator
+  def initialize
+    connection
+  end
+
+  def zip_file
+    File.open(tempfile)
+  end
+
+  def configset_exists?
+    resp = @connection.get('/solr/admin/configs?action=LIST&omitHeader=true')
+    JSON.parse(resp.body)['configSets'].include?(configset_name)
+  end
+
+  def collection_exists?
+    resp = @connection.get('/solr/admin/collections?action=LIST')
+    JSON.parse(resp.body)['collections'].include?(collection_name)
+  end
+
+  def create_collection
+    if collection_exists?
+      return true
+    end
+
+    resp = @connection.get("/solr/admin/collections?action=CREATE&name=#{collection_name}&numShards=#{num_shards}")
+    check_resp(resp)
+  end
+
+  def modify_collection
+    puts 'Modifying Collection to use Configset'
+    uri = %W(
+      /solr/admin/collections
+      ?action=MODIFYCOLLECTION
+      &collection=#{collection_name}
+      &collection.configName=#{@configset_name}
+    ).join
+    resp = @connection.get(uri)
+    check_resp(resp)
+  end
+
+  def upload_config
```

Comment:
To me, all these puts statements below seem to mix the concerns of this class with the concerns of the script that will be running it. What if I wanted to use this class outside the context of a rake task / script? What if this were part of some kind of web UI for configuring Solr instances instead?

I could see an argument for moving most of this logic into the rake task itself instead, since `configset_exists?` is a private method, and that would be a more appropriate . The main plus for keeping this logic together in the method is, it makes it easier to test the idempotency...except that there currently is no test for that.